.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,27 @@
+name: pre-commit hooks
+
+on:
+  - push
```

Comment:
[optional] some people prefer running hooks only on PR to remove runs doubling, we can try it too